tests/main/snap-remove-terminate/task.yaml

```
@@ -22,16 +26,23 @@ execute: |
     echo "Start a long running process"
     lockfile="$(pwd)/lockfile"
     touch "$lockfile"
-    sh_snap_bin="$(command -v test-snapd-sh.sh)"
-    systemd-run --unit test-kill.service flock "$lockfile" "$sh_snap_bin" -c 'touch /var/snap/test-snapd-sh/common/alive; sleep 100000'
+
+    sh_snap_bin="$(command -v fork-bomb.sh)"
+    if [ "$BAD_SNAP" = "fork-bomb" ]; then
+        #shellcheck disable=SC2016
+        systemd-run --unit test-kill.service flock "$lockfile" "$sh_snap_bin" -c 'touch /var/snap/test-snapd-sh/common/alive; $SNAP/bin/fork-bomb'
```

so that it doesn't blow up too much in the tests:
```suggestion
        systemd-run --unit test-kill.service -p TasksMax=1000 flock "$lockfile" "$sh_snap_bin" -c 'touch /var/snap/test-snapd-sh/common/alive; $SNAP/bin/fork-bomb'
```

---

It seems to only apply to the outer systemd-run, but snapd's systemd-run escapes the outer cgroup limits imposed by TasksMax.

Interestingly, I think we accidentally just discovered that snaps don't behave normally when run as part of systemd directly because `snap run` calls systemd-run which create a separate unit/cgroup.

---

Yeah, that's unfortunate. Maybe this should be treated as a bug really, filed https://warthogs.atlassian.net/browse/SNAPDENG-32298 feel free to grab it